downloadrepos

```diff
@@ -13,25 +15,98 @@ downloadgithubrepos() {
     repo_owner="`echo "$owner_and_repo_name" | sed "s@/.*\\$@@g"`"
     repo_name="`echo "$owner_and_repo_name" | sed "s@^.*/@@g"`"
     repo_branch="$1"; shift
-    set -x
     # clean up other previously downloaded branches of the same repo as well
     rm -rf ${repo_name}-*
     ls | grep $repo_name
     downloadrepos https://github.com/$repo_owner/$repo_name "$repo_branch"
     ls | grep $repo_name
-    set +x
 }
 
-. ./default_build_params
+# USAGE: VAR_TO_LOWER="$(lowercase "$VAR_TO_LOWER")"
+lowercase() {
+    echo "$1" | tr '[:upper:]' '[:lower:]'
+}
+
+lowercase_boolean_build_params() {
+    TEST_MAGPIE_AUTH="$(lowercase "$TEST_MAGPIE_AUTH")"
+    TEST_PAVICS_SDI_REPO="$(lowercase "$TEST_PAVICS_SDI_REPO")"
+    TEST_PAVICS_SDI_WEAVER="$(lowercase "$TEST_PAVICS_SDI_WEAVER")"
+    TEST_FINCH_REPO="$(lowercase "$TEST_FINCH_REPO")"
+    TEST_PAVICS_LANDING_REPO="$(lowercase "$TEST_PAVICS_LANDING_REPO")"
+    TEST_RAVEN_REPO="$(lowercase "$TEST_RAVEN_REPO")"
+    TEST_RAVENPY_REPO="$(lowercase "$TEST_RAVENPY_REPO")"
+    TEST_ESGF_COMPUTE_API_REPO="$(lowercase "$TEST_ESGF_COMPUTE_API_REPO")"
+    TEST_LOCAL_NOTEBOOKS="$(lowercase "$TEST_LOCAL_NOTEBOOKS")"
+}
+
+# Replace all slash (/) by dash (-) because (/) is illegal in folder name
+# for branch name of the format "feature/my_wizbang-feature".
+# Github does the same when downloading repo archive by downloadrepos above.
+# USAGE: export BRANCH_NAME="$(sanitize_branch_name "$BRANCH_NAME")"
+sanitize_branch_name() {
+    echo "$1" | sed "s@/@-@g"
+}
+
+# Ex: extract 'pavics-sdi' from 'Ouranosinc/pavics-sdi'.
+# USAGE: REPO_NAME_ONLY="$(extract_repo_name "$REPO_NAME")"
+extract_repo_name() {
+    echo "$1" | sed "s@^.*/@@g"
+}
+
+# Branches that have allowed characters such as '+' other than alphanum, '-', '_' and '.' are converted to '-' in archives.
+# USAGE: FOLDER_NAME="$(sanitize_extracted_folder_name "$FOLDER_NAME")"
+sanitize_extracted_folder_name() {
+    echo "$1" | sed "s@[^a-zA-Z0-9_\-\.]@-@g"
+}
+
+# Presence of setup.cfg, tox.ini, pyproject.toml files confuse py.test execution rootdir discovery.
+# USAGE: delete_files_confusing_pytest "$CHECKOUT_DIR"
+delete_files_confusing_pytest() {
+    for afile in setup.cfg tox.ini pyproject.toml; do
+        if [ -f "$1/$afile" ]; then
+            rm -v "$1/$afile"
+        fi
+    done
+}
```

Comment:
If the repository defined pytest options, such as necessary plugins or marker definitions, this will break their execution.

Reply:
In that case we should be running a separate `py.test` process for each repo instead of all at once. 

I agree with @fmigneault that there could be crucial definitions in the `setup.cfg tox.ini pyproject.toml` files that we do not want to lose or the tests may fail unexpectedly.